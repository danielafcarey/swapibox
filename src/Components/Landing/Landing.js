import React from 'react';
import PropTypes from 'prop-types';

const Landing = ({ crawlData }) => {
  const { title, releaseDate, crawlText } = crawlData;
  return (
    <div className="Landing">
      <div className="crawl">
        <p>{ crawlText }</p>
        <div className="title">
          <h1>{ title }</h1>
          <p>{ releaseDate }</p>
        </div>
      </div>
    </div>
  );
};

Landing.propTypes = {
  crawlData: PropTypes.shape({
    title: PropTypes.string,
    releaseDate: PropTypes.string,
    crawlText: PropTypes.string
  })
};

export default Landing;
