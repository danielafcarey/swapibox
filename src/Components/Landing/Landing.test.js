import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';

describe('Landing', () => {

  it('matches the snapshot', () => {
    const mockCrawlData = {
      crawlText: 'i am a movie', 
      title: 'moviem',
      releaseDate: 'today'
    }
    const wrapper = shallow(<Landing crawlData={ mockCrawlData } />)

    expect(wrapper).toMatchSnapshot();
  });

});
