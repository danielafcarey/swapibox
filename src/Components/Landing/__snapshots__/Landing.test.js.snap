// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    crawlData={
        Object {
            "crawlText": "i am a movie",
            "releaseDate": "today",
            "title": "moviem",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="crawl"
>
        <p>
                i am a movie
        </p>
        <div
                className="title"
        >
                <h1>
                        moviem
                </h1>
                <p>
                        today
                </p>
        </div>
</div>,
      "className": "Landing",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            i am a movie
</p>,
          <div
            className="title"
>
            <h1>
                        moviem
            </h1>
            <p>
                        today
            </p>
</div>,
        ],
        "className": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "i am a movie",
          },
          "ref": null,
          "rendered": "i am a movie",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                moviem
</h1>,
              <p>
                today
</p>,
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "moviem",
              },
              "ref": null,
              "rendered": "moviem",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "today",
              },
              "ref": null,
              "rendered": "today",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <p>
                    i am a movie
          </p>
          <div
                    className="title"
          >
                    <h1>
                              moviem
                    </h1>
                    <p>
                              today
                    </p>
          </div>
</div>,
        "className": "Landing",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              i am a movie
</p>,
            <div
              className="title"
>
              <h1>
                            moviem
              </h1>
              <p>
                            today
              </p>
</div>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "i am a movie",
            },
            "ref": null,
            "rendered": "i am a movie",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  moviem
</h1>,
                <p>
                  today
</p>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "moviem",
                },
                "ref": null,
                "rendered": "moviem",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "today",
                },
                "ref": null,
                "rendered": "today",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
